modules/thermal_hydraulics/test/tests/problems/brayton_cycle/closed_brayton_cycle.i

```diff
@@ -0,0 +1,607 @@
+  []
+  [motor_power_fn]
+    type = ParsedFunction
+    value = 'torque * speed'
+    vars = 'torque speed'
+    vals = 'motor_torque shaft:omega'
+  []
+  [generator_torque_fn]
+    type = ParsedFunction
+    value = 'slope * t'
+    vars = 'slope'
+    vals = '${generator_torque_per_shaft_speed}'
+  []
+  [generator_power_fn]
+    type = ParsedFunction
+    value = 'torque * speed'
+    vars = 'torque speed'
+    vals = 'generator_torque shaft:omega'
+  []
+  [htc_wall_fn]
+    type = PiecewiseLinear
+    x = '0 ${t1} ${t2}'
+    y = '0 0 1e3'
+  []
+[]
+
+[Modules/FluidProperties]
+  [fp_air]
+    type = IdealGasFluidProperties
+    emit_on_nan = none
+  []
+[]
+
+[Closures]
+  [closures]
+    type = Closures1PhaseSimple
+  []
+[]
+
+[Components]
+  [shaft]
+    type = Shaft
+    connected_components = 'motor compressor turbine generator'
+    initial_speed = ${speed_initial}
+  []
+  [motor]
+    type = ShaftConnectedMotor
+    inertia = ${I_motor}
+    torque = 0 # controlled
+  []
+  [generator]
+    type = ShaftConnectedMotor
+    inertia = ${I_generator}
+    torque = generator_torque_fn
+  []
+
+  [pipe1]
+    type = FlowChannel1Phase
+    position = '${x1} 0 0'
+    length = ${L1}
+    n_elems = ${n_elems1}
+    A = ${A1}
+  []
+  [compressor]
+    type = ShaftConnectedCompressor1Phase
+    position = '${x2} 0 0'
+    inlet = 'pipe1:out'
+    outlet = 'pipe2:in'
+    A_ref = ${A_ref_comp}
+    volume = ${V_comp}
+
+    omega_rated = ${speed_rated}
+    mdot_rated = ${rated_mfr}
+    c0_rated = ${c0_rated_comp}
+    rho0_rated = ${rho0_rated_comp}
+
+    speeds = '0.5208 0.6250 0.7292 0.8333 0.9375'
+    Rp_functions = 'rp_comp1 rp_comp2 rp_comp3 rp_comp4 rp_comp5'
+    eff_functions = 'eff_comp1 eff_comp2 eff_comp3 eff_comp4 eff_comp5'
+
+    min_pressure_ratio = 1.0
+
+    speed_cr_I = 0
+    inertia_const = ${I_comp}
+    inertia_coeff = '${I_comp} 0 0 0'
+
+    # assume no shaft friction
+    speed_cr_fr = 0
+    tau_fr_const = 0
+    tau_fr_coeff = '0 0 0 0'
+  []
+  [pipe2]
+    type = FlowChannel1Phase
+    position = '${x2} 0 0'
+    length = ${L2}
+    n_elems = ${n_elems2}
+    A = ${A2}
+  []
+  [junction2_3]
+    type = JunctionOneToOne1Phase
+    connections = 'pipe2:out pipe3:in'
+  []
+  [pipe3]
+    type = FlowChannel1Phase
+    position = '${x3} 0 0'
+    length = ${L3}
+    n_elems = ${n_elems3}
+    A = ${A3}
+  []
+  [junction3_4]
+    type = JunctionOneToOne1Phase
+    connections = 'pipe3:out pipe4:in'
+  []
+  [pipe4]
+    type = FlowChannel1Phase
+    position = '${x4} 0 0'
+    length = ${L4}
+    n_elems = ${n_elems4}
+    A = ${A4}
+  []
+  [turbine]
+    type = ShaftConnectedCompressor1Phase
+    position = '${x5} 0 0'
+    inlet = 'pipe4:out'
+    outlet = 'pipe5:in'
+    A_ref = ${A_ref_turb}
+    volume = ${V_turb}
+
+    treat_as_turbine = true
+
+    omega_rated = ${speed_rated}
+    mdot_rated = ${rated_mfr}
+    c0_rated = ${c0_rated_comp}
+    rho0_rated = ${rho0_rated_comp}
```

Comment:
The rated density would probably be different from the compressor as the inlet temperature and pressures are higher. But this is ok for this demonstration purpose.